File: types/api/keys.ts
export interface CreateApiKeyRequest {
  label: string;
  expires_at?: string | null;
}

export interface CreateApiKeyResponse {
  id: string;
  label: string;
  key: string;
  prefix: string;
  created_at: string;
  expires_at: string | null;
  is_active: boolean;
}

export interface ApiKeyInfo {
  id: string;
  label: string;
  prefix: string;
  created_at: string;
  expires_at: string | null;
  is_active: boolean;
  last_used_at: string | null;
}

export interface ListApiKeysResponse {
  keys: ApiKeyInfo[];
}

export interface RotateApiKeyResponse {
  id: string;
  new_key: string;
}

export interface RevokeApiKeyResponse {
  id: string;
  revoked: boolean;
}

File: app/api/schemas/keys.py
from datetime import datetime

from pydantic import BaseModel, Field


class CreateApiKeyRequest(BaseModel):
    label: str = Field(min_length=1, max_length=100, description="Human-readable label for the key")
    expires_at: datetime | None = None
    organization_id: int = Field(description="Organization to scope this key to")


class CreateApiKeyResponse(BaseModel):
    id: str
    label: str
    key: str
    prefix: str
    created_at: datetime
    expires_at: datetime | None = None
    is_active: bool


class ApiKeyInfo(BaseModel):
    id: str
    label: str
    prefix: str
    created_at: datetime
    expires_at: datetime | None = None
    is_active: bool
    last_used_at: datetime | None = None


class ListApiKeysResponse(BaseModel):
    keys: list[ApiKeyInfo]


class RotateApiKeyResponse(BaseModel):
    id: str
    new_key: str


class RevokeApiKeyResponse(BaseModel):
    id: str
    revoked: bool